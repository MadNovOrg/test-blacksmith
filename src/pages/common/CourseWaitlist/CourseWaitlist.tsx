import { Box, Typography, Link, Alert } from '@mui/material'
import React, { useRef } from 'react'
import { useTranslation } from 'react-i18next'
import { useSearchParams } from 'react-router-dom'
import { useMutation, useQuery } from 'urql'

import {
  JoinWaitlistMutation,
  JoinWaitlistMutationVariables,
  WaitlistCourseQuery,
  WaitlistCourseQueryVariables,
} from '@app/generated/graphql'
import { AppLayoutMinimal } from '@app/layouts/AppLayoutMinimal'

import { CourseInfo, CourseInfoSkeleton } from './components/CourseInfo'
import { Form, FormInputs } from './components/Form'
import { JoinedWaitlist } from './components/JoinedWaitlist'
import { JOIN_WAITLIST, WAITLIST_COURSE } from './queries'

export const CourseWaitlist: React.FC<
  React.PropsWithChildren<unknown>
> = () => {
  const { t } = useTranslation()
  const [searchParams] = useSearchParams()
  const courseId = searchParams.get('course_id')
  const emailRef = useRef('')

  const [{ data, fetching }] = useQuery<
    WaitlistCourseQuery,
    WaitlistCourseQueryVariables
  >({
    query: WAITLIST_COURSE,
    variables: { id: Number(courseId) ?? 0 },
  })

  const [
    {
      data: joinedWaitlistData,
      fetching: joiningWaitlist,
      error: errorJoinWaitlist,
    },
    joinWaitlistMutation,
  ] = useMutation<JoinWaitlistMutation, JoinWaitlistMutationVariables>(
    JOIN_WAITLIST
  )

  const course = data?.courses.length ? data.courses[0] : null

  const joinWaitlist = (data: FormInputs) => {
    emailRef.current = data.email ?? ''
    joinWaitlistMutation({
      input: {
        courseId: Number(courseId),
        givenName: data.firstName,
        familyName: data.surname,
        phone: data.phone,
        email: data.email,
        orgName: data.orgName,
        recaptchaToken: data.recaptchaToken,
      },
    })
  }

  if (joinedWaitlistData?.joinWaitlist.success && emailRef.current) {
    return <JoinedWaitlist email={emailRef.current} />
  }

  const courseFound = course && !fetching

  return (
    <AppLayoutMinimal
      width={628}
      contentBoxStyles={{ p: 3 }}
      footer={
        <Box mt={4}>
          <Link
            href={`mailto:${import.meta.env.VITE_TT_INFO_EMAIL_ADDRESS}`}
            target="_blank"
            rel="noopener"
            fontWeight={600}
            color="primary"
          >
            {t('need-help')}? {t('contact-us')}
          </Link>
        </Box>
      }
    >
      <Box
        display="flex"
        flexDirection="column"
        alignItems="center"
        data-testid="join-waiting-list-form"
      >
        <Typography
          variant="h3"
          fontWeight="600"
          color="secondary"
          gutterBottom
        >
          {t('join-waitlist-title')}
        </Typography>
        {courseFound ? (
          <Typography
            variant="body1"
            textAlign="center"
            color="grey.700"
            mb={4}
          >
            {t('join-waitlist-notice')}
          </Typography>
        ) : null}

        {fetching && !course ? <CourseInfoSkeleton /> : null}

        {course ? <CourseInfo course={course} /> : null}
      </Box>

      {!course && !fetching ? (
        <Alert variant="outlined" severity="error" sx={{ mt: 2 }}>
          {t('waitlist-not-found')}
        </Alert>
      ) : null}

      {errorJoinWaitlist ? (
        <Alert variant="outlined" severity="error" sx={{ mt: 2 }}>
          {t('waitlist.join-error')}
        </Alert>
      ) : null}

      {course && !fetching ? (
        <Form onSuccess={joinWaitlist} saving={joiningWaitlist} />
      ) : null}
    </AppLayoutMinimal>
  )
}
