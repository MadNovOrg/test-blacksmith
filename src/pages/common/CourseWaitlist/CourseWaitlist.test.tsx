import React from 'react'
import { MemoryRouter, Route, Routes } from 'react-router-dom'

import { render } from '@test/index'

import { CourseWaitlist } from './CourseWaitlist'

describe('CourseWaitlistPage', () => {
  it('matches snapshot', async () => {
    const view = render(
      <MemoryRouter initialEntries={['/waitlist?course_id=10000']}>
        <Routes>
          <Route path="/waitlist" element={<CourseWaitlist />} />
        </Routes>
      </MemoryRouter>
    )

    expect(view).toMatchSnapshot()
  })
})
