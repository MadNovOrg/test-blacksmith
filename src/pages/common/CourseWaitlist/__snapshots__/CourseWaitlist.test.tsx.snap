// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseWaitlistPage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-16d3e73"
      >
        <div
          class="MuiBox-root css-70qvj9"
        >
          <div
            height="80"
            width="80"
          />
        </div>
        <div
          class="MuiBox-root css-1wlfb0w"
        >
          <div
            class="MuiBox-root css-dvxtzn"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1qu2ikj-MuiTypography-root"
            >
              Join the course waiting list
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-12xmzka-MuiTypography-root"
            >
              The course you are interested in is currently full. We will notify you via email if a space becomes available.
            </p>
          </div>
          <form
            aria-autocomplete="none"
            autocomplete="off"
            class="MuiBox-root css-178yklu"
            novalidate=""
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18lu0av-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-myzee5-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jgv580-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-rl2i38-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="firstName"
                    id="firstName-label"
                  >
                    First Name
                    <span
                      aria-hidden="true"
                      class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-1nkheju-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      data-testid="input-first-name"
                      id="firstName"
                      name="firstName"
                      placeholder=""
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-myzee5-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jgv580-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1n83x2o-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="surname"
                    id="surname-label"
                  >
                    Surname
                    <span
                      aria-hidden="true"
                      class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1nkheju-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      data-testid="input-surname"
                      id="surname"
                      name="surname"
                      placeholder=""
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-i3pbo"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jgv580-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1n83x2o-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1nkheju-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    data-testid="input-email"
                    id="email"
                    name="email"
                    placeholder=""
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-10ib5jr"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1m0bcqz-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rl2i38-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-1"
                  id="mui-1-label"
                >
                  Phone
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-1nkheju-MuiInputBase-root-MuiInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium MuiPhoneNumber-positionStart css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <button
                      aria-haspopup="true"
                      aria-label="Select country"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiPhoneNumber-flagButton css-1h7flgg-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        class="margin"
                        viewBox="0 85.333 512 341.333"
                      >
                        <path
                          d="M0 85.333h512V426.67H0z"
                          fill="#FFF"
                        />
                        <path
                          d="M288 85.33h-64v138.666H0v64h224v138.666h64V287.996h224v-64H288z"
                          fill="#D80027"
                        />
                        <path
                          d="M393.785 315.358L512 381.034v-65.676zm-82.133 0L512 426.662v-31.474l-143.693-79.83zm146.982 111.304l-146.982-81.664v81.664z"
                          fill="#0052B4"
                        />
                        <path
                          d="M311.652 315.358L512 426.662v-31.474l-143.693-79.83z"
                          fill="#FFF"
                        />
                        <path
                          d="M311.652 315.358L512 426.662v-31.474l-143.693-79.83z"
                          fill="#D80027"
                        />
                        <path
                          d="M90.341 315.356L0 365.546v-50.19zm110.007 14.154v97.151H25.491z"
                          fill="#0052B4"
                        />
                        <path
                          d="M143.693 315.358L0 395.188v31.474l200.348-111.304z"
                          fill="#D80027"
                        />
                        <path
                          d="M118.215 196.634L0 130.958v65.676zm82.133 0L0 85.33v31.474l143.693 79.83zM53.366 85.33l146.982 81.664V85.33z"
                          fill="#0052B4"
                        />
                        <path
                          d="M200.348 196.634L0 85.33v31.474l143.693 79.83z"
                          fill="#FFF"
                        />
                        <path
                          d="M200.348 196.634L0 85.33v31.474l143.693 79.83z"
                          fill="#D80027"
                        />
                        <path
                          d="M421.659 196.636L512 146.446v50.19zm-110.007-14.154V85.331h174.857z"
                          fill="#0052B4"
                        />
                        <path
                          d="M368.307 196.634L512 116.804V85.33L311.652 196.634z"
                          fill="#D80027"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1bsm28j-MuiInputBase-input-MuiInput-input"
                    data-testid="input-phone"
                    id="mui-1"
                    placeholder="+1 (702) 123-4567"
                    required=""
                    type="tel"
                    value="+44"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-i3pbo"
            >
              <div
                class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-11riny3-MuiAutocomplete-root"
                data-testid="org-selector"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1n83x2o-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="mui-3"
                    id="mui-3-label"
                  >
                    Organisation Name
                    <span
                      aria-hidden="true"
                      class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1nkheju-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      id="mui-3"
                      placeholder="Organisation name"
                      required=""
                      role="combobox"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-cbvi0e-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-dvxtzn"
            >
              <button
                class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-14srt4-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                data-testid="btn-submit"
                id="mui-4"
                tabindex="0"
                type="submit"
              >
                Join Waiting List
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-16d3e73"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <div
          height="80"
          width="80"
        />
      </div>
      <div
        class="MuiBox-root css-1wlfb0w"
      >
        <div
          class="MuiBox-root css-dvxtzn"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1qu2ikj-MuiTypography-root"
          >
            Join the course waiting list
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-12xmzka-MuiTypography-root"
          >
            The course you are interested in is currently full. We will notify you via email if a space becomes available.
          </p>
        </div>
        <form
          aria-autocomplete="none"
          autocomplete="off"
          class="MuiBox-root css-178yklu"
          novalidate=""
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18lu0av-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-myzee5-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jgv580-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-rl2i38-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="firstName"
                  id="firstName-label"
                >
                  First Name
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-1nkheju-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    data-testid="input-first-name"
                    id="firstName"
                    name="firstName"
                    placeholder=""
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-myzee5-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jgv580-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1n83x2o-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="surname"
                  id="surname-label"
                >
                  Surname
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1nkheju-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    data-testid="input-surname"
                    id="surname"
                    name="surname"
                    placeholder=""
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-i3pbo"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-jgv580-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1n83x2o-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="email"
                id="email-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1nkheju-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  data-testid="input-email"
                  id="email"
                  name="email"
                  placeholder=""
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-10ib5jr"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1m0bcqz-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rl2i38-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-1"
                id="mui-1-label"
              >
                Phone
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-1nkheju-MuiInputBase-root-MuiInput-root"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium MuiPhoneNumber-positionStart css-ittuaa-MuiInputAdornment-root"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                  <button
                    aria-haspopup="true"
                    aria-label="Select country"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiPhoneNumber-flagButton css-1h7flgg-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      class="margin"
                      viewBox="0 85.333 512 341.333"
                    >
                      <path
                        d="M0 85.333h512V426.67H0z"
                        fill="#FFF"
                      />
                      <path
                        d="M288 85.33h-64v138.666H0v64h224v138.666h64V287.996h224v-64H288z"
                        fill="#D80027"
                      />
                      <path
                        d="M393.785 315.358L512 381.034v-65.676zm-82.133 0L512 426.662v-31.474l-143.693-79.83zm146.982 111.304l-146.982-81.664v81.664z"
                        fill="#0052B4"
                      />
                      <path
                        d="M311.652 315.358L512 426.662v-31.474l-143.693-79.83z"
                        fill="#FFF"
                      />
                      <path
                        d="M311.652 315.358L512 426.662v-31.474l-143.693-79.83z"
                        fill="#D80027"
                      />
                      <path
                        d="M90.341 315.356L0 365.546v-50.19zm110.007 14.154v97.151H25.491z"
                        fill="#0052B4"
                      />
                      <path
                        d="M143.693 315.358L0 395.188v31.474l200.348-111.304z"
                        fill="#D80027"
                      />
                      <path
                        d="M118.215 196.634L0 130.958v65.676zm82.133 0L0 85.33v31.474l143.693 79.83zM53.366 85.33l146.982 81.664V85.33z"
                        fill="#0052B4"
                      />
                      <path
                        d="M200.348 196.634L0 85.33v31.474l143.693 79.83z"
                        fill="#FFF"
                      />
                      <path
                        d="M200.348 196.634L0 85.33v31.474l143.693 79.83z"
                        fill="#D80027"
                      />
                      <path
                        d="M421.659 196.636L512 146.446v50.19zm-110.007-14.154V85.331h174.857z"
                        fill="#0052B4"
                      />
                      <path
                        d="M368.307 196.634L512 116.804V85.33L311.652 196.634z"
                        fill="#D80027"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1bsm28j-MuiInputBase-input-MuiInput-input"
                  data-testid="input-phone"
                  id="mui-1"
                  placeholder="+1 (702) 123-4567"
                  required=""
                  type="tel"
                  value="+44"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-i3pbo"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-11riny3-MuiAutocomplete-root"
              data-testid="org-selector"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1n83x2o-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="mui-3"
                  id="mui-3-label"
                >
                  Organisation Name
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-aihfuz-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1nkheju-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="mui-3"
                    placeholder="Organisation name"
                    required=""
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-cbvi0e-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-dvxtzn"
          >
            <button
              class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-14srt4-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              data-testid="btn-submit"
              id="mui-4"
              tabindex="0"
              type="submit"
            >
              Join Waiting List
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "context": Object {
    "auth": Object {
      "activeRole": "user",
      "allowedRoles": Set {
        "user",
      },
      "changeRole": [MockFunction],
      "defaultRole": "user",
      "getJWT": [MockFunction],
      "loadProfile": [MockFunction],
      "loading": false,
      "login": [MockFunction],
      "logout": [MockFunction],
      "profile": Object {
        "addresses": Array [],
        "attributes": Array [],
        "avatar": "",
        "contactDetails": Array [],
        "createdAt": "",
        "dietaryRestrictions": null,
        "disabilities": null,
        "dob": "2002-04-06",
        "email": "john.smith@example.com",
        "familyName": "Smith",
        "fullName": "John Smith",
        "givenName": "John",
        "id": "cacb559d-b85d-5e64-b623-37252520ebda",
        "jobTitle": "Developer",
        "lastActivity": 2020-01-01T00:00:00.000Z,
        "organizations": Array [],
        "phone": "+44 1234555",
        "preferences": Array [],
        "roles": Array [
          Object {
            "role": Object {
              "id": "1",
              "name": "user",
            },
          },
        ],
        "tags": Array [],
        "title": "Mr",
        "updatedAt": "",
      },
      "reloadCurrentProfile": [MockFunction],
    },
  },
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
